os/board/rtl8721csm/src/component/common/api/wifi_interactive_mode.c

```diff
@@ -488,6 +490,15 @@ int8_t cmd_wifi_connect(trwifi_ap_config_s *ap_connect_config, void *arg)
 		return -1;
 	}
 
+	if ((ap_channel >= 1) && (ap_channel <= 13)) {
+		pscan_config = PSCAN_ENABLE | PSCAN_FAST_SURVEY;
+		ret = wifi_set_pscan_chan((uint8_t *)&ap_channel, &pscan_config, 1);
+
+		if (ret < 0) {
+			RTW_API_INFO("\n\rset pscan failed");

```

Comment:
Is it okay not to return as fail like `-1`?

Reply:
The `wifi_set_pscan_chan()` set the partial scan and driver scans only that channels set by this API.
If this API returns '-1' due to some reason, it only means driver will not do partial scan and continue with full scan.
So we don't really have to stop connect wifi just because `wifi_set_pscan_chan()` is failed.